src/tools/perf/lib/libperf.c

```diff
@@ -1271,12 +1272,44 @@ static void ucp_perf_test_destroy_workers(ucx_perf_context_t *perf)
     }
 }
 
-void ucx_perf_set_warmup(ucx_perf_context_t* perf,
-                         const ucx_perf_params_t* params)
+ucs_status_t
+ucx_perf_do_warmup(ucx_perf_context_t *perf, const ucx_perf_params_t *params)
 {
-    perf->max_iter = ucs_min(params->warmup_iter,
-                             ucs_div_round_up(params->max_iter, 10));
-    perf->report_interval = ULONG_MAX;
+    ucs_time_t deadline = ucs_get_time() +
+                          ucs_time_from_sec(params->warmup_time);
+    ucx_perf_counter_t warmup_iter, total_warmup_iter;
+    ucs_status_t status, stop_status;
+
+    /* Perform no more than 'params->warmup_iter' iterations but try to not
+       exceed 'params->warmup_time' */
+    warmup_iter       = 1;
+    total_warmup_iter = 0;
```

Comment:
why is this var needed?

Reply:
total_warmup_iter is to count total number of iteration we did for warmup, and stop when it reached params->warmup_iter

Reply:
ic, it's used in `while`